src/site/apt/index.apt.vm

```diff
@@ -40,6 +40,13 @@ ${project.name}
   General instructions on how to use the GPG Plugin can be found on the {{{./usage.html}usage page}}. Some more
   specific use cases are described in the examples given below.
 
+  By default, plugin will enforce "best practices", and will fail the build if any violation are detected.
+  In short, this was done to stop users putting secrets (plaintext or quasi-encrypted) in their Maven configuration
+  files (settings.xml, POMs) or use secrets in a way they leave trace (like in terminal history). In this default mode,
+  plugin leaves two options to obtain passphrase: use of gpg-agent in interactive sessions, and use of environment
+  variables in batch/non-interactive sessions. To disable "best practices" and regain full backward compatibility,
+  configure the plugin accordingly (see goals, look for <<<bestPractices>>> configuration).
```

Comment:
What about CI on non-docker environments? Other users on the system will be able to read the process' environment variables, which is worse than having the password in a configuration file.
"best practices" really depends on your environment.

Reply:
[Environment variables](https://en.wikipedia.org/wiki/Environment_variable)

Reply:
And no, you would never ever export any env variable (or set on root process) that contains sensitive info, you usually set it on process that needs it (and is per def inherited by childs, if forking). But this way, once the process finishes, the env variable is gone with it.